File: Count and Say.cpp
class Solution {
public:
    string countAndSay(int n) {
        string currentSequence;
        for(int i = 0; i < n; ++i){
            string newSequence;
            if(i == 0){
                newSequence = "1";
            }
            else{
                for(int i = 0; i < (int) currentSequence.size();){
                    int j = i;
                    while(j < (int) currentSequence.size() and currentSequence[j] == currentSequence[i] and ++j);
                    newSequence += to_string(j - i);
                    newSequence.push_back(currentSequence[i]);
                    i = j;
                }
            }
            currentSequence = newSequence;
        }
        return currentSequence;
    }
};

File: Count and Say.java
class Solution {
    public String countAndSay(int n) {
        HashMap <Integer, String> countAndSay = new HashMap <>();
        countAndSay.put(1, "1");
        for(int term = 2; term <= n; ++term){
            String previous = countAndSay.get(term - 1);
            String current = new String();
            for(int index = 0; index < previous.length();){
                int sameTillIndex = index;
                while(sameTillIndex < previous.length() && previous.charAt(sameTillIndex) == previous.charAt(index)){
                    ++sameTillIndex;
                }
                current += Integer.toString(sameTillIndex - index);
                current += previous.charAt(index);
                index = sameTillIndex;
            }
            countAndSay.put(term, current);
        }
        return countAndSay.get(n);
    }
}
